src/scripts/move.js

```diff
@@ -0,0 +1,230 @@
+
+  for (const cell of allCells) {
+    if (cell.classList.length === 1) {
+      emptyFields.push(cell);
+    }
+  }
+
+  // create new item:
+  if (emptyFields.length > 0) {
+    const randomIndex = randomField(0, emptyFields.length - 1);
+    const randomValue = randomField(0, values.length - 1);
+
+    emptyFields[randomIndex].style.opacity = '0';
+    emptyFields[randomIndex].style.transition = 'opacity 0.2s';
+
+    emptyFields[randomIndex]
+      .classList.add('field-cell--' + values[randomValue]);
+
+    emptyFields[randomIndex]
+      .innerText = values[randomValue];
+
+    setTimeout(() => {
+      emptyFields[randomIndex].style.opacity = '1';
+      emptyFields[randomIndex].removeAttribute('style');
+    }, 100);
+
+  // if its impossible to create a new item:
+  } else {
+    messageAfterLose.classList.remove('hidden');
+    messageDuringTheGame.classList.add('hidden');
+
+    setNewHighestScore(gameScore, highestScore);
+
+    // mainListenerRemover();
+    document.body.removeEventListener('keyup', arrowsPlayHandler);
+  }
+}
+
+// Main movement Handler:
+function arrowsPlayHandler(e) {
+  switch (e.key) {
+    case 'ArrowDown':
+      mainMove({
+        mainAxis: 0,
+        staticAxis: 1,
+        revert: false,
+        shouldReverse: true,
+      });
+      break;
+
+    case 'ArrowUp':
+      mainMove({
+        mainAxis: 0,
+        staticAxis: 1,
+        revert: true,
+        shouldReverse: false,
+      });
+      break;
+
+    case 'ArrowLeft':
+      mainMove({
+        mainAxis: 1,
+        staticAxis: 0,
+        revert: true,
+        shouldReverse: false,
+      });
+      break;
+
+    case 'ArrowRight':
+      mainMove({
+        mainAxis: 1,
+        staticAxis: 0,
+        revert: false,
+        shouldReverse: true,
+      });
+      break;
+  }
+}
+
+// Main movement algorithm
+function mainMove({
+  mainAxis,
+  staticAxis,
+  revert,
+  shouldReverse,
+}) {
+  let isMoved = false;
+  const notEmptyFields = getNotEmptyFields(allCells);
+  const notEmptyFieldsPositions = getNotEmptyFieldsCoords(allCells);
+  const allGameCells = [...allCells];
+
+  if (shouldReverse) {
+    notEmptyFields.reverse();
+    allGameCells.reverse();
+  }
+
+  for (const field of notEmptyFields) {
+    const fieldPos = field.id.split('-').slice(1);
+
+    for (const cell of allGameCells) {
+      const cellPos = cell.id.split('-').slice(1);
+
+      if (checkForMove(
+        field, cell, fieldPos, cellPos,
+        mainAxis, staticAxis, revert
+      )) {
+        moveCell(field, cell, fieldPos, cellPos, mainAxis);
+        isMoved = true;
+        break;
+      } else if (
+        checkForMerge(
+          field, cell, fieldPos, cellPos,
+          mainAxis, staticAxis, revert
+        )
+        && checkBetween(
+          fieldPos, cellPos, notEmptyFieldsPositions,
+          mainAxis, revert
+        )
+      ) {
+        mergeCells(field, cell, fieldPos, cellPos, mainAxis);
+        isMoved = true;
+        break;
+      }
```

Comment:
```suggestion
      const canMove = checkForMove(
        field, cell, fieldPos, cellPos,
        mainAxis, staticAxis, revert
      );
      const canMerge = checkForMerge(
        field, cell, fieldPos, cellPos,
        mainAxis, staticAxis, revert
      );
      const isBetween = checkBetween(
        fieldPos, cellPos, notEmptyFieldsPositions,
        mainAxis, revert
      );

      if (canMove) {
        moveCell(field, cell, fieldPos, cellPos, mainAxis);
        isMoved = true;
        break;
      } 
      
      if (canMerge && isBetween)  { // since you've used break before
        mergeCells(field, cell, fieldPos, cellPos, mainAxis);
        isMoved = true;
        break;
      }
```

Reply:
Let's move this checks result to variable, it's will be much easier to understand conditions